src/uucore/src/lib/mods/error.rs

```diff
@@ -0,0 +1,198 @@
+use std::{
+    error::Error,
+    fmt::{Display, Formatter},
+    path::PathBuf,
+    sync::atomic::{AtomicI32, Ordering},
+};
+
+static EXIT_CODE: AtomicI32 = AtomicI32::new(0);
+
+pub fn get_exit_code() -> i32 {
+    EXIT_CODE.load(Ordering::SeqCst)
+}
+
+/// Set the exit code for the program if `uumain` returns `Ok(())`.
+pub fn set_exit_code(code: i32) {
+    EXIT_CODE.store(code, Ordering::SeqCst);
```

Comment:
Do we want to add some logic here to make sure we don't call it multiple times with different exit codes? Maybe it's a clear indication of a bug if we first store one exit code and then another, but I don't know.

Reply:
Maybe, but I think it's too early to say. I would have to look at some more utils to see how this might be used.